packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -222,31 +237,35 @@ export class PipelineEditor extends React.Component<
       {
         action: 'export',
         label: 'Export Pipeline',
-        enable: true,
+        enable: !this.state.emptyPipeline,
         iconEnabled: IconUtil.encode(exportPipelineIcon),
         iconDisabled: IconUtil.encode(exportPipelineIcon)
       },
       {
         action: 'clear',
         label: 'Clear Pipeline',
-        enable: true,
+        enable: !this.state.emptyPipeline,
         iconEnabled: IconUtil.encode(clearPipelineIcon),
         iconDisabled: IconUtil.encode(clearPipelineIcon)
       },
       { divider: true },
-      { action: 'undo', label: 'Undo', enable: true },
-      { action: 'redo', label: 'Redo', enable: true },
-      { action: 'cut', label: 'Cut', enable: false },
-      { action: 'copy', label: 'Copy', enable: false },
+      { action: 'undo', label: 'Undo', enable: !this.state.emptyPipeline },
+      { action: 'redo', label: 'Redo', enable: !this.state.emptyPipeline },
```

Comment:
These probably should not be disabled on empty pipeline since the user might want to redo/undo adding or deleting the only node in the canvas.

When testing locally I noticed that they don't actually get disabled/enabled as I would have expected based on this line. They behave as I would expect them to (can still undo redo based on the "clipboard" not pipeline state) but when undoing or redoing the first/last node in the canvas the undo/redo doesn't trigger an update to whether the other buttons should be enabled/disabled. ie they stay in whichever state they were in prior to the undo/redo

Reply:
<img width="543" alt="Screen Shot 2020-07-09 at 1 44 04 PM" src="https://user-images.githubusercontent.com/13952758/87088828-4ad1e880-c1ea-11ea-9106-75a8b389cef4.png">

Reply:
Good catch, gonna take a look

Reply:
`undo/redo` is fixed, also similar cases noticed with `delete` and `addComment`. Note we want `clear pipeline` enabled if there are comment boxes but no nodes.
`cut/paste` already trigger model update, didn't need to add listeners for those (in case we cut the only node left, or paste a node on an empty pipeline)